src/libstd/dynamic_lib.rs

```diff
@@ -263,6 +266,16 @@ mod dl {
                  symbol: *const libc::c_char) -> *mut libc::c_void;
         fn dlclose(handle: *mut libc::c_void) -> libc::c_int;
     }
+
+    #[cfg(target_os = "openbsd")]
```

Comment:
Since `#[link_name="dl"]` is actually a no-op on the first block, I think you could remove the second one, the `cfg` and take off the `#[link_name]` attribute from the first block.